slides/07_conclusion.qmd

```diff
@@ -0,0 +1,100 @@
+---
+title: "Conclusion"
+subtitle: "BBS Course: Good Software Engineering Practice for R Packages"
+author: "Daniel"
+date: '2023-02-10'
+date-format: long
+tags: [R, Package]
+license: 'CC BY-SA'
+email-obfuscation: javascript
+format: 
+  revealjs: 
+    fontsize: 32px
+    footer: 'Good Software Engineering Practice for R Packages - [Home](../index.html "GSWEP4R Workshop") | [Slides](listing.html "Overview of all slides") | [License](http://creativecommons.org/licenses/by-sa/4.0/ "License: CC BY-SA 4.0")'
+description: "Discussion of learning experience, summary of workflow, conclusion and next steps.</br>[Photo by Ann H on pexels.com]{.copyright}"
+image: thumbnails/conclusion.jpg
+---
+
+# Temperature check
+
+## What have you learned?
+
+\[slido wordcloud here\]
+
+Plus also ask folks to speak up in the room directly
+
+## What can you apply in your work now?
+
+\[slido wordcloud here\]
+
+Plus also ask folks to speak up in the room directly
+
+## What was especially difficult?
+
+\[slido wordcloud here\]
+
+Plus also ask folks to speak up in the room directly
+
+# Take away messages
+
+## R package structure
+
+-   R package = folder structure with (many) conventions
+-   With modern tools `usethis`, `roxygen2` it is easy to start a new package
+-   Packaging a set of functions is an ideal way to share with collaborators and the public
+-   Start small and simple and over time you can learn additional options
+
+## Engineering Workflow
+
+-   Use a workable workflow:\
+    Idea $\rightarrow$ Design docs $\rightarrow$ Programming $\rightarrow$ Quality check $\rightarrow$ Publication
+    -   Don't waste time on maintenance
+    -   Be faster with release on CRAN
+    -   Fulfill regulatory requirements
+-   Refactor your one-off scripts that you want to use in a package
+-   Use assertions for all arguments $\rightarrow$ Usability/UX
+-   Implement common generics like `print` and `plot`
+
+## Ensuring Quality
+
+-   Apply common clean code rules, e.g.,
+    -   Use [testthat](https://cran.r-project.org/package=testthat) to test, test, and test
+    -   Use [covr](https://cran.r-project.org/package=covr) to improve the test coverage
+    -   Use [styler](https://styler.r-lib.org) to optimize the code styling
+    -   Don't repeat yourself!
+-   Your rewards are:
+    -   [{{< fa trophy size=1x >}}]{.yellow} **Maintainability**,
+    -   [{{< fa trophy size=1x >}}]{.yellow} **Extensibility**, and
+    -   [{{< fa trophy size=1x >}}]{.yellow} **Performance**!
+
+## Collaboration
+
```

Comment:
@kkmann can you add here the collaboration take aways?

Reply:
done